File: day-4/solve2.py
import sys
import math

lines = sys.stdin.readlines()  # 2-4,4-6

total = 0
for line in lines:
    lhs, rhs = line.split(",")
    # construct lhs range
    idx = lhs.index("-")
    idxr = rhs.index("-")
    lhs_range = range(int(lhs[0:idx]), int(lhs[idx + 1 :]) + 1)
    rhs_range = range(int(rhs[0:idxr]), int(rhs[idxr + 1 :]) + 1)
    for n in lhs_range:
        if n in rhs_range:
            total += 1
            break
print(total)

File: day-4/solve2.jl
lines = eachline()

total = 0
for line in lines
  r1, r2 = split(line, ',')
  a, b = split(r1, '-') |> x -> parse.(Int, x)
  c, d = split(r2, '-') |> x -> parse.(Int, x)
  if (a >= c && a <= d) || (b >= c && b <= d) || (c >= a && c <= b) || (d >= a && d <= b)
    global total += 1
  end
end

total |> println
